web-common/src/features/dashboards/time-dimension-details/Pivot.svelte

```diff
@@ -0,0 +1,216 @@
+<script lang="ts">
+  import { createVirtualizer, VirtualItem } from "@tanstack/svelte-virtual";
+  import PivotVirtualRow from "./PivotVirtualRow.svelte";
+  import PivotCell from "./PivotCell.svelte";
+
+  export let rowCt: number;
+  export let colCt: number;
+  export let fixedColCt: number;
+  export let getColumnWidth: (idx: number) => number;
+  export let getRowSize: (idx: number) => number;
+  export let renderCell: (rowIdx: number, colIdx: number) => any;
+  export let height: number;
+
+  function range(n: number) {
+    return new Array(n).fill(0).map((d, i) => i);
+  }
+
+  let container: HTMLDivElement;
+  $: rowVirtualizer = createVirtualizer({
+    count: rowCt,
+    getScrollElement: () => container,
+    estimateSize: getRowSize,
+    overscan: 10,
+  });
+
+  let virtualRows: VirtualItem[] = [];
+  let totalVerticalSize = 0;
+  let paddingTop = 0;
+  let paddingBottom = 0;
+  $: {
+    virtualRows = $rowVirtualizer?.getVirtualItems() ?? [];
+    totalVerticalSize = $rowVirtualizer?.getTotalSize() ?? 0;
+    paddingTop = virtualRows?.length > 0 ? virtualRows?.[0]?.start || 0 : 0;
+    paddingBottom =
+      virtualRows?.length > 0
+        ? totalVerticalSize - (virtualRows?.at(-1)?.end || 0)
+        : 0;
+  }
+
+  $: columnVirtualizer = createVirtualizer({
+    horizontal: true,
+    count: colCt,
+    getScrollElement: () => container,
+    estimateSize: getColumnWidth,
+    overscan: 5,
+  });
+
+  let totalHorizontalSize = 0;
+  let paddingLeft = 0;
+  let paddingRight = 0;
+  let fixedVirtualColumns: VirtualItem[] = [];
+  let nonfixedVirtualColumns: VirtualItem[] = [];
+  $: {
+    const virtualColumns = $columnVirtualizer?.getVirtualItems() ?? [];
+    totalHorizontalSize = $columnVirtualizer?.getTotalSize() ?? 0;
+
+    // Manually calculate fixed virtual column set as they may not be in current virtualized item set
+    fixedVirtualColumns = range(fixedColCt).reduce((arr, idx) => {
+      const start = arr[idx - 1]?.end ?? 0;
+      const size = getColumnWidth(idx);
+      const end = start + getColumnWidth(idx);
+      return [
+        ...arr,
+        {
+          index: idx,
+          start,
+          end,
+          key: idx,
+          lane: 0,
+          size,
+        },
+      ];
+    }, []);
+
+    // If current virtual column set has fixed columns, remove them since we will use our measurements
+    nonfixedVirtualColumns = virtualColumns.filter(
+      (c) => c.index >= fixedColCt
+    );
+
+    paddingLeft =
```

Comment:
Feels weird to set paddingLeft and set it again in the next line. Perhaps use a local variable for this line?